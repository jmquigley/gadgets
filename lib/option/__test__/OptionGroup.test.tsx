'use strict';

import {mount, shallow} from 'enzyme';
import * as React from 'react';
import {getDefaultOptionGroupProps, OptionGroup} from '../index';

test('Test retrieval of OptionGroup props object', () => {
	const props = getDefaultOptionGroupProps();

	expect(props).toBeTruthy();
	expect(props).toMatchSnapshot();
});

test('Creation of the OptionGroup control', () => {
	const ctl = shallow(
		<OptionGroup
			className="test-class"
			default="option1"
			options={[
				'option1',
				'option2',
				'option3'
			]}
			title="test options"
		/>
	);

	expect(ctl).toBeTruthy();
	expect(ctl).toMatchSnapshot();
});

// TODO: add test for disabling, invisible, and click/select event
