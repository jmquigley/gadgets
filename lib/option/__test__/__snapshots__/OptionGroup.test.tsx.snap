// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creation of the OptionGroup control 1`] = `
<ThemeProvider
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "color": "#000000",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.5s",
    }
  }
>
  <styled.div
    className="ui-option-group test-class"
  >
    <Styled(Title)
      className="ui-option-group-title"
      layout={6}
      noedit={true}
      noripple={true}
      title="test options"
    />
    <Styled(Option)
      key="option1"
      onClick={[Function]}
      optionType={0}
      selected={true}
      text="option1"
    />
    <Styled(Option)
      key="option2"
      onClick={[Function]}
      optionType={0}
      selected={false}
      text="option2"
    />
    <Styled(Option)
      key="option3"
      onClick={[Function]}
      optionType={0}
      selected={false}
      text="option3"
    />
  </styled.div>
</ThemeProvider>
`;

exports[`Test retrieval of OptionGroup props object 1`] = `
Object {
  "bottom": "",
  "children": null,
  "className": "",
  "contentEditable": false,
  "default": "",
  "direction": "right",
  "disabled": false,
  "errorMessage": "",
  "focus": false,
  "height": "",
  "id": "",
  "left": "",
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "obj": "OptionGroup",
  "onSelect": [Function],
  "optionType": 0,
  "options": Array [],
  "padding": "",
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "title": "",
  "tooltip": "",
  "top": "",
  "visible": true,
  "width": "",
  "xcss": null,
}
`;
